File: src/main/java/tk/hiddenname/smarthome/exception/UnsupportedSignalTypeException.java
package tk.hiddenname.smarthome.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;
import tk.hiddenname.smarthome.entity.signal.SignalType;

@ResponseStatus(value = HttpStatus.CONFLICT)
public class UnsupportedSignalTypeException extends RuntimeException {

    public UnsupportedSignalTypeException(SignalType type) {
        super(String.format("Signal type \"%s\" exists but not supported by your current device's configuration",
                type.toString()));
    }
}

File: src/main/java/tk/hiddenname/smarthome/exception/UnsupportedSignalTypeException.kt
package tk.hiddenname.smarthome.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus
import tk.hiddenname.smarthome.model.signal.SignalType

@ResponseStatus(value = HttpStatus.CONFLICT)
class UnsupportedSignalTypeException(type: SignalType) :
        RuntimeException("Signal type \"$type\" exists but not supported by your current device's configuration")